From 53da21574da32dc54f4dc91c4b46fd817f30837f Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 19 Sep 2024 16:01:18 +0200
Subject: [PATCH] dt-bindings: mailbox: Add char device mailbox client bindings

Add devicetree binding for the mailbox character device that exposes
a /dev/mailbox<%x> interface to communicate with a remote processor,
relying on a shared memory region and a mailbox channel.

Change-Id: Ic7c1b40d76ad280913f2160558925e0f6d811076
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/405532
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 .../bindings/mailbox/mbox-cdev.yaml           | 55 +++++++++++++++++++
 1 file changed, 55 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/mailbox/mbox-cdev.yaml

diff --git a/Documentation/devicetree/bindings/mailbox/mbox-cdev.yaml b/Documentation/devicetree/bindings/mailbox/mbox-cdev.yaml
new file mode 100644
index 000000000000..a7986bcdf9e3
--- /dev/null
+++ b/Documentation/devicetree/bindings/mailbox/mbox-cdev.yaml
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mailbox/mbox-cdev.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Mailbox Character Device
+
+description: |
+  This binding describes a mailbox character device that uses a mailbox
+  framework and a memory region to communicate with a remote processor.
+
+maintainers:
+  - Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
+
+properties:
+  compatible:
+    const: mbox-cdev
+
+  memory-region:
+    description: |
+      Phandle to the reserved memory region used for communication with the
+      coprocessor.
+    $ref: /schemas/types.yaml#/definitions/phandle
+
+  mboxes:
+    description: |
+      phandles to the mailbox channel used for communication.
+    maxItems: 1
+
+required:
+  - compatible
+  - memory-region
+  - mboxes
+
+additionalProperties: false
+
+examples:
+  - |
+    reserved-memory {
+      #address-cells = <1>;
+      #size-cells = <1>;
+      ranges;
+
+      memory_region: memory-region@20000000 {
+        reg = <0x20000000 0x10000>;
+        no-map;
+      };
+    };
+
+    mbox-cdev {
+      compatible = "mbox-cdev";
+      memory-region = <&memory_region>;
+      mboxes = <&mailbox0>;
+    };
-- 
2.39.5

